google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -350,6 +354,32 @@ private OutstandingMessage(ReceivedMessage receivedMessage, AckHandler ackHandle
     }
   }
 
+  private static class ReceiptCompleteData {
+    private OutstandingMessage outstandingMessage;
+    private Boolean receiptComplete;
+
+    private ReceiptCompleteData(OutstandingMessage outstandingMessage, Boolean receiptComplete) {
+      this.outstandingMessage = outstandingMessage;
+      this.receiptComplete = receiptComplete;
+    }
+
+    private OutstandingMessage getOutstandingMessage() {
+      return this.outstandingMessage;
+    }
+
+    private Boolean getReceiptComplete() {
+      return this.receiptComplete;
+    }
+
+    private void setOutstandingMessage(OutstandingMessage outstandingMessage) {
+      this.outstandingMessage = outstandingMessage;
+    }
+
+    private void setReceiptComplete(Boolean receiptComplete) {
```

Comment:
We should never set receiptComplete to false after creation of this object. Let's just have this method be called notifyReceiptComplete(), and have the implementation set receiptComplete to false.

Reply:
i assume you mean set receiptComplete to true here? This method is used when we call notifyAckSuccess(), so we need to set receipt complete to true in this method so that we can process the message.

Reply:
Yes correct, I just meant we should never set receiptComplete to false, so we shouldn't allow the user to provide the value of receiptComplete. And it looks like you made this change already, so this is resolved.